lib/modules/platform/gitea/utils.ts

```diff
@@ -7,3 +16,44 @@ export function smartLinks(body: string): string {
 export function trimTrailingApiPath(url: string): string {
   return url?.replace(regEx(/api\/v1\/?$/g), '');
 }
+
+export function getRepoUrl(
+  repo: helper.Repo,
+  gitUrl: GitUrlOption | undefined,
+  endpoint: string
+): string {
+  if (gitUrl === 'ssh') {
+    if (!repo.ssh_url) {
+      throw new Error(CONFIG_GIT_URL_UNAVAILABLE);
+    }
+    logger.debug({ url: repo.ssh_url }, `using SSH URL`);
+    return repo.ssh_url;
+  }
+
+  // Find options for current host and determine Git endpoint
+  const opts: HostRuleSearchResult = hostRules.find({
+    hostType: PlatformId.Gitea,
+    url: endpoint,
+  });
+
+  if (gitUrl === 'endpoint') {
+    const { protocol, host, pathname } = parseUrl(endpoint) ?? {};
+    const url = URL.format({
+      protocol: protocol?.slice(0, -1) || 'https',
+      auth: opts.token,
+      host,
+      pathname: pathname + repo.full_name + '.git',
+    });
+    logger.debug({ url }, 'using URL based on configured endpoint');
+    return url;
+  }
+
+  if (!repo.clone_url) {
+    throw new Error(CONFIG_GIT_URL_UNAVAILABLE);
+  }
+
+  logger.debug({ url: repo.clone_url }, `using HTTP URL`);
+  const repoUrl = URL.parse(`${repo.clone_url}`);
```

Comment:
```suggestion
  const repoUrl = URL.parse(repo.clone_url);
```
as we've now validate, this string workaround is no longer required.